pkg/frontend/querymiddleware/roundtrip.go

```diff
@@ -308,12 +307,19 @@ func isInstantQuery(path string) bool {
 	return strings.HasSuffix(path, instantQueryPathSuffix)
 }
 
-func defaultInstantQueryParamsRoundTripper(next http.RoundTripper, now func() time.Time) http.RoundTripper {
+func defaultInstantQueryParamsRoundTripper(next http.RoundTripper) http.RoundTripper {
 	return RoundTripFunc(func(r *http.Request) (*http.Response, error) {
 		if isInstantQuery(r.URL.Path) && !r.URL.Query().Has("time") {
+			nowUnixStr := strconv.FormatInt(time.Now().Unix(), 10)
+
 			q := r.URL.Query()
-			q.Add("time", strconv.FormatInt(time.Now().Unix(), 10))
+			q.Add("time", nowUnixStr)
 			r.URL.RawQuery = q.Encode()
+
+			// If form was already parsed, add this param to the form too.
```

Comment:
We should also mention that if form already had `time` parameter, this extra `time` value will be ignored.

Reply:
Sounds good. Thanks.